cvmfs/download.cc

```diff
@@ -177,25 +177,27 @@ static size_t CallbackCurlHeader(void *ptr, size_t size, size_t nmemb,
   //          header_line.c_str());
 
   // Check http status codes
-  if (HasPrefix(header_line, "HTTP/1.", false)) {
-    if (header_line.length() < 10)
+  if (HasPrefix(header_line, "HTTP/", false)) {
+    if (header_line.length() < 12) {
```

Comment:
Actually the HTTP/2 header is `HTTP/2.0 [...]`, didn't we figure this out while implementing this patch? The old check was simply wrong...

`HTTP/1.1 200` is 12 characters to begin with. But looking into the history of the code a few lines below, it used to check only the first digit of the HTTP status code. When it was extended to check the second and third digit as well, this sanity check was just not updated.